sdk/resourcemanager/azure-resourcemanager-samples/src/main/java/com/azure/resourcemanager/compute/samples/CloneVirtualMachineToNewRegion.java

```diff
@@ -0,0 +1,314 @@
+                    .withRegion(regionNew)
+                    .withExistingResourceGroup(rgNameNew)
+                    .withData()
+                    .fromSnapshot(dataSnapshot.id())
+                    .create();
+                newDataDisks.add(dataDisk);
+
+                System.out.println("Created managed disk holding data: " + dataDisk.id());
+                // ResourceManagerUtils.print(dataDisk); TODO
+                i++;
+            }
+
+            //
+            //=============================================================
+            // Create a Linux VM by attaching the managed disks
+
+            System.out.println("Creating a Linux VM using specialized OS and data disks");
+
+            VirtualMachine linuxVM2 = azureResourceManager.virtualMachines().define(linuxVMName2)
+                .withRegion(regionNew)
+                .withExistingResourceGroup(rgNameNew)
+                .withNewPrimaryNetwork("10.0.0.0/28")
+                .withPrimaryPrivateIPAddressDynamic()
+                .withoutPrimaryPublicIPAddress()
+                .withSpecializedOSDisk(newOSDisk, OperatingSystemTypes.LINUX)
+                .withExistingDataDisk(newDataDisks.get(0))
+                .withExistingDataDisk(newDataDisks.get(1), 1, CachingTypes.READ_WRITE)
+                .withSize(VirtualMachineSizeTypes.fromString("Standard_D2a_v4"))
+                .create();
+
+            Utils.print(linuxVM2);
+
+            //=============================================================
+            //
+            System.out.println("Deleting OS snapshot - " + osSnapshotNewRegion.id());
+
+            azureResourceManager.snapshots().deleteById(osSnapshotNewRegion.id());
+
+            System.out.println("Deleted OS snapshot");
+
+            for (Snapshot dataSnapshot : dataSnapshots) {
+                System.out.println("Deleting data snapshot - " + dataSnapshot.id());
+
+                azureResourceManager.snapshots().deleteById(dataSnapshot.id());
+
+                System.out.println("Deleted data snapshot");
+            }
+
+            System.out.println("De-allocating the virtual machine - " + linuxVM2.id());
+
+            linuxVM2.deallocate();
+            return true;
+        } finally {
+            try {
+                System.out.println("Deleting Resource Group: " + rgName);
+                azureResourceManager.resourceGroups().beginDeleteByName(rgName);
+                System.out.println("Deleted Resource Group: " + rgName);
+                System.out.println("Deleting Resource Group: " + rgNameNew);
+                azureResourceManager.resourceGroups().beginDeleteByName(rgNameNew);
+                System.out.println("Deleted Resource Group: " + rgNameNew);
+            } catch (NullPointerException npe) {
+                System.out.println("Did not create any resources in Azure. No clean up is necessary");
```

Comment:
why?

Reply:
If resource group create failed, NPE will be thrown.

Reply:
eh, why?

And that does not means NPE = resource group create failed.
NPE can be from other places.

Reply:
Right, makes sense.